hull3/acre_functions.sqf

```diff
@@ -193,56 +176,63 @@ hull3_acre_fnc_getSideStep = {
 };
 
 hull3_acre_fnc_setRadioChannels = {
-    FUN_ARGS_1(_unit);
-
-    private ["_defaultShortRangeChannel", "_shortRangeChannelAssignments", "_shortRangeRadios", "_defaultLongRangeChannel", "_longRangeChannelAssignments"];
-    _defaultShortRangeChannel = ["ACRE", "ShortRange", "defaultChannel"] call hull3_config_fnc_getNumber;
-    _shortRangeChannelAssignments = ["ACRE", "ShortRange", "channels"] call hull3_config_fnc_getBothArray;
-    _shortRangeRadios = ["ACRE", "ShortRange", "radios"] call hull3_config_fnc_getArray;
-    _defaultLongRangeChannel = ["ACRE", "LongRange", "defaultChannel"] call hull3_config_fnc_getNumber;
-    _longRangeChannelAssignments = ["ACRE", "LongRange", "channels"] call hull3_config_fnc_getBothArray;
-    {
-        private ["_channelAssignments", "_defaultChannel", "_channel"];
-        if (([_x] call acre_api_fnc_getBaseRadio) in _shortRangeRadios) then {
-            _channelAssignments = _shortRangeChannelAssignments;
-            _defaultChannel = _defaultShortRangeChannel;
-        } else {
-            _channelAssignments = _longRangeChannelAssignments;
-            _defaultChannel = _defaultLongRangeChannel;
-        };
-        _channel = [_unit, _channelAssignments, _defaultChannel] call hull3_acre_fnc_getRadioChannelFromGroupId;
-        [_x, _channel] call acre_api_fnc_setRadioChannel;
-    } foreach ([] call acre_api_fnc_getCurrentRadioList);
-    ["acre.channels.set", [player]] call hull3_event_fnc_emitEvent;
+    params ["_unit"];
+
+    [
+        {(_this #0) getVariable ["hull3_gear_radiosAssigned", false] && {call acre_api_fnc_isInitialized} },
```

Comment:
FYI, you could have used `gear.radio.assigned` Hull event. But since you would still need to wait for ACRE, it would make the code less readable, so it is fine as is.

Reply:
Done